src/WSRequestHandler_ComponentsManagement.cpp

```diff
@@ -0,0 +1,145 @@
+* @param {Object} `hotkeyData` hotkeyData. These will be merged to the new created source.
+*
+* @api requests
+* @name CreateNewSource
+* @category sources
+*/
+HandlerResponse WSRequestHandler::HandleCreateNewSource(WSRequestHandler* req)
+{
+	if (!req->hasField("sourceName") || !req->hasField("sourceType") || !req->hasField("sourceSettings")) {
+		return req->SendErrorResponse("missing request parameters");
+	}
+	const char* sourceName = obs_data_get_string(req->data, "sourceName");
+	const char* sourceType = obs_data_get_string(req->data, "sourceType");
+	
+	OBSDataAutoRelease newsourceSettings = obs_data_get_obj(req->data, "sourceSettings");
+	OBSDataAutoRelease hotkeyData = obs_data_get_obj(req->data, "hotkeyData");
+	
+	obs_source_t* source = obs_source_create(sourceType, sourceName, newsourceSettings, hotkeyData);
+	obs_source_set_enabled(source, true);
+
+	OBSDataAutoRelease response = obs_data_create();
+	obs_data_set_string(response, "sourceName", obs_source_get_name(source));
+	obs_data_set_string(response, "sourceType", obs_source_get_id(source));
+	obs_data_set_obj(response, "sourceSettings", newsourceSettings);
+	obs_data_set_obj(response, "hotkeyData", hotkeyData);
+
+	if (source == nullptr) {
+		return req->SendErrorResponse("Could not create a new source");	
+	}
+	return req->SendOKResponse(response);
+}
+
+/**
+* Create new scene
+*
+* @param {String} `sceneName` Name of the Scene to be created
+*
+* @api requests
+* @name CreateNewScene
+* @category sources
+*/
+HandlerResponse WSRequestHandler::HandleCreateNewScene(WSRequestHandler* req)
+{
+	if (!req->hasField("sceneName") ) {
+		return req->SendErrorResponse("missing request parameters");
+	}
+
+	const char* sceneName = obs_data_get_string(req->data, "sceneName");
+
+	obs_scene_t* scene = obs_scene_create(sceneName);
+	OBSDataAutoRelease response = obs_data_create();
+	obs_data_set_string(response, "sceneName", sceneName);
+	if (scene == nullptr) {
+		return req->SendErrorResponse("Could not create a new scene");
+	}
+	return req->SendOKResponse(response);
+}
+
+/**
+* Add active child
+*
+* @param {String} `sourceParentName` Name of the parent source
+* @param {String} `sourceChildName` Name of the child source
+*
+* @api requests
+* @name AddActiveChild
+* @category sources
+*/
+HandlerResponse WSRequestHandler::HandleAddActiveChild(WSRequestHandler* req)
+{
+	if (!req->hasField("sourceParentName") || !req->hasField("sourceChildName")) {
+		return req->SendErrorResponse("missing request parameters");
+	}
+	const char* sourceParentName = obs_data_get_string(req->data, "sourceParentName");
+	const char* sourceChildName = obs_data_get_string(req->data, "sourceChildName");
+	obs_source_t* sourceParent= obs_get_source_by_name(sourceParentName);
+	obs_source_t* sourceChild = obs_get_source_by_name(sourceChildName);
+	
+	bool  addingresponse =  obs_source_add_active_child(sourceParent, sourceChild);
```

Comment:
I believe you shouldn't use `obs_source_add_active_child` directly. Instead, scenes and groups use dedicated scene item functions for this.